### PR TITLE
Update Sentry & Snuba

### LINKED ISSUE #78: Attempting to update to latest sentry tag caused Internal errors for sentry
I attempted to use latest tags for sentry and snuba:
`getsentry/sentry:00ae12b89402f10158e36413d35a94d642b56082`
`getsentry/snuba:3b11c3b64901d91fbbd0622ca718bcb7d871dd4a`
But once rolled out, sentry became unusable, with every page giving internal error. 

`snuba-api`, `snuba-consumer` were logging this error:
`clickhouse_driver.errors.NetworkError: Code: 210. Cannot assign requested address (localhost:9000)`
I wonder if something changed with regards to how they expect clickhouse address to be provided? Since it shouldn't be using `localhost` to connect to clickhouse.

@J0sh0nat0r you mentioned previously that you are running using latest tags for sentry? Did you do anything differently on your end to make it work?

This is using chart version 1.3.1

### PR DESCRIPTION
**Pull request by J0sh0nat0r**
This PR updates the default Sentry & Snuba tags, and applies a compatible configuration.

Fixes #78

### CODE REVIEW

File: sentry/templates/hooks/clickhouse-init.job.yaml [hunk 1]
```diff
@@ -38,7 +39,9 @@ spec:
           - -ec
           - >-
             for ((i=0;i<{{ .Values.clickhouse.clickhouse.replicas }};i++)); do
+              clickhouse-client --database=default --host={{ $clickhouseHost }}-$i.{{ $clickhouseHost }}-headless --port={{ $clickhousePort }} --query='{{ $dropQuery }}';
```
Comment (Mokto) [inline sentry/templates/hooks/clickhouse-init.job.yaml]: Does it mean we drop the table every time we run the hooks?

Comment (J0sh0nat0r) [reply sentry/templates/hooks/clickhouse-init.job.yaml]: Yes, however the dustributed table only contains the structure and a reference to the local table where the data is actually stored so no data loss occurs.

Comment (Mokto) [reply sentry/templates/hooks/clickhouse-init.job.yaml]: Thanks for the info! 

I'll try it really soon.

**Comment by Mokto** [comment]
I just tried it!
I don't have the issue, but I don't see any event anymore, when I click on an issue : "No events found".

Snuba API logs seem normal : 
```
10.48.0.211 - - [10/Jun/2020:13:43:38 +0000] "POST /query HTTP/1.1" 200 403 "tagstore.get_groups_user_counts" "-"
10.48.0.211 - - [10/Jun/2020:13:43:38 +0000] "POST /query HTTP/1.1" 200 588 "tagstore.get_group_seen_values_for_environments" "-"
10.48.0.211 - - [10/Jun/2020:13:43:39 +0000] "POST /query HTTP/1.1" 200 356 "tagstore.__get_release" "-"
10.48.0.211 - - [10/Jun/2020:13:43:39 +0000] "POST /query HTTP/1.1" 200 581 "tagstore.__get_tag_keys" "-"
10.48.0.211 - - [10/Jun/2020:13:43:39 +0000] "POST /query HTTP/1.1" 200 1640 "tsdb" "-"
10.48.0.211 - - [10/Jun/2020:13:43:39 +0000] "POST /query HTTP/1.1" 200 1581 "tsdb" "-"
10.48.0.211 - - [10/Jun/2020:13:43:39 +0000] "POST /query HTTP/1.1" 200 403 "tagstore.get_groups_user_counts" "-"
10.48.0.211 - - [10/Jun/2020:13:43:39 +0000] "POST /query HTTP/1.1" 200 588 "tagstore.get_group_seen_values_for_environments" "-"
10.48.0.211 - - [10/Jun/2020:13:43:40 +0000] "POST /query HTTP/1.1" 200 581 "tagstore.__get_tag_keys" "-"
10.48.0.211 - - [10/Jun/2020:13:43:40 +0000] "POST /query HTTP/1.1" 200 356 "tagstore.__get_release" "-"
10.48.0.211 - - [10/Jun/2020:13:43:40 +0000] "POST /query HTTP/1.1" 200 3397 "tagstore.__get_tag_keys_and_top_values" "-"
10.48.0.211 - - [10/Jun/2020:13:43:40 +0000] "POST /query HTTP/1.1" 200 582 "tagstore.__get_tag_keys" "-"
10.48.0.211 - - [10/Jun/2020:13:43:40 +0000] "POST /query HTTP/1.1" 200 1590 "tsdb" "-"
10.48.0.211 - - [10/Jun/2020:13:43:40 +0000] "POST /query HTTP/1.1" 200 1533 "tsdb" "-"
10.48.0.211 - - [10/Jun/2020:13:43:40 +0000] "POST /query HTTP/1.1" 200 596 "Group.get_latest" "-"
10.48.0.211 - - [10/Jun/2020:13:43:40 +0000] "POST /query HTTP/1.1" 200 339 "eventstore.get_event_by_id_nodestore" "-"
```

Also, my events page is really weird:

Screenshot 2020-06-10 at 15 44 34

**Comment by J0sh0nat0r** [comment]
@Mokto That's rather strange, is there anything untoward going on in your `suba-consumer` / `snuba-replacer` logs?

**Comment by Mokto** [comment]
sentry-web:
```
06:37:02 [WARNING] sentry.eventstore.snuba.backend: eventstore.missing-snuba-event (event_id=u'5349068a25564ac1a93b2c3ba0fb66c5' project_id=4L)
06:37:02 [WARNING] django.request: Not Found: /api/0/issues/2264/events/latest/ (status_code=404 request=<WSGIRequest: GET u'/api/0/issues/2264/e
vents/latest/'>)
10.48.0.126 - - [11/Jun/2020:06:37:02 +0000] "GET /api/0/issues/2264/events/latest/ HTTP/1.1" 404 543 "https://sentry.anchor.io/organizations/oce
an/issues/2264/?query=is%3Aunresolved" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.9
7 Safari/537.36"
10.48.0.126 - - [11/Jun/2020:06:37:02 +0000] "POST /api/1/store/?sentry_key=d772d7e362714875893a76e8b5ba102b&sentry_version=7 HTTP/1.1" 200 703 "
https://sentry.anchor.io/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537
.36"
```

snuba-replacer
```
2020-06-11 01:43:43,067 New partitions assigned: {Partition(topic=Topic(name='event-replacements'), index=0): 0}
```

snuba-consumer
```
2020-06-11 06:39:51,546 Worker flush took 51ms
2020-06-11 06:39:59,382 Flushing 3 items (from {Partition(topic=Topic(name='events'), index=0): Offsets(lo=585912, hi=585914)}): forced:False siz
e:False time:True
2020-06-11 06:39:59,428 Worker flush took 45ms
2020-06-11 06:40:18,908 Flushing 1 items (from {Partition(topic=Topic(name='events'), index=0): Offsets(lo=585915, hi=585915)}): forced:False siz
e:False time:True
2020-06-11 06:40:19,048 Worker flush took 138ms
2020-06-11 06:40:35,359 Flushing 4 items (from {Partition(topic=Topic(name='events'), index=0): Offsets(lo=585916, hi=585919)}): forced:False siz
e:False time:True
2020-06-11 06:40:35,422 Worker flush took 60ms
2020-06-11 06:40:36,863 Flushing 1 items (from {Partition(topic=Topic(name='events'), index=0): Offsets(lo=585920, hi=585920)}): forced:False siz
e:False time:True
2020-06-11 06:40:36,918 Worker flush took 54ms
```

So no, nothing wrong here, I think.

My configuration is quite classic too:

```

  kafka:
    enabled: true
    metrics:
      kafka:
        enabled: true
      serviceMonitor:
        enabled: true

  clickhouse:
    clickhouse:
      persistentVolumeClaim:
        enabled: true
        dataPersistentVolume: 
          enabled: true
          accessModes: 
          - "ReadWriteOnce"
          storageClassName: "standard"
          storage: "30Gi"

  filestore:
    backend: gcs
    gcs:
      credentialsFile: credentials.json
      secretName: sentry-gcs
      bucketName: oceanio-sentry
    
    filesystem:
      persistence:
        enabled: false
```

Put 1 replica of every service (to debug) + mail, slack, githubSso, github

**Comment by Mokto** [comment]
Are your databases (kafka, clickhouse) run through the chart ?

**Comment by Mokto** [comment]
@J0sh0nat0r Btw i just gave you write access on the repo

**Comment by Mokto** [comment]
Ok, I tried something.

I tried to match with the new docker-compose file that Sentry has done : https://github.com/getsentry/onpremise/blob/master/docker-compose.yml

So I basically (starting point is this branch) :
1. Updated snuba consumer deployment by adding `--storage events` in the command.
1. Updated snuba replacer deployment by adding `--storage events` in the command.
2. Added snuba-outcomes-consumers deployment.
3. Added snuba-session-consumers deployment.

Now outcomes consumer is returning:

```
Traceback (most recent call last):
  File "/usr/local/bin/snuba", line 11, in <module>
    load_entry_point('snuba', 'console_scripts', 'snuba')()
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/src/snuba/snuba/cli/consumer.py", line 162, in consumer
    consumer.run()
  File "/usr/src/snuba/snuba/utils/streams/batching.py", line 137, in run
    self._run_once()
  File "/usr/src/snuba/snuba/utils/streams/batching.py", line 142, in _run_once
    self._flush()
  File "/usr/src/snuba/snuba/utils/streams/batching.py", line 242, in _flush
    self.worker.flush_batch(self.__batch_results)
  File "/usr/src/snuba/snuba/consumer.py", line 109, in flush_batch
    self.__writer.write(inserts)
  File "/usr/src/snuba/snuba/clickhouse/http.py", line 84, in write
    raise ClickhouseError(int(code), message)
snuba.clickhouse.errors.ClickhouseError: [60] Table default.outcomes_raw_dist doesn't exist. (version 19.16.19.85 (official build))
```

**Comment by Mokto** [comment]
Sorry for spamming.

But I guess we'll need to add 
https://getsentry.github.io/relay
to the project, also.

**Comment by Mokto** [comment]
Added the same table initialisation queries that you've made (without really understanding how this works).

```
{{- $dropQueryOutcomes := "DROP TABLE IF EXISTS outcomes_raw_dist" -}}
{{- $createQueryOutcomes := .Release.Name | printf "CREATE TABLE outcomes_raw_dist AS outcomes_raw_local ENGINE = Distributed(%s-clickhouse, default, outcomes_raw_local, rand())" -}}
```

Now everything fine, on paper.

I also tried to make some queries to snuba api directly :

```
{
    "project":[17],
    "aggregations": [["count()", "", "event_count"]]
}
```

Returns 
```
{
    "data": [
        {
            "event_count": 1021
        }
    ],
    "meta": [
        {
            "name": "event_count",
            "type": "UInt64"
        }
    ],
    "timing": {
        "timestamp": 1591865475,
        "duration_ms": 30,
        "marks_ms": {
            "cache_get": 0,
            "cache_set": 2,
            "execute": 16,
            "get_configs": 0,
            "prepare_query": 5,
            "rate_limit": 2,
            "validate_schema": 2
        }
    }
}
```



I'm starting to wonder if it could be because of https://github.com/getsentry/snuba/issues/965

**Comment by Mokto** [comment]
Another message :D 

I put 

```
timezone: "Europe/London"
```

On the clickhouse chart, restart the masters&replicas.
And now I can see events in the events pages.

But still have the same issue on individual events. events

Screenshot 2020-06-11 at 11 57 21

**Comment by J0sh0nat0r** [comment]
@Mokto Happy to collaborate, thanks for the access :)

Yes I'm running both Kafka and Clickhouse via the chart, and I think you may right about the timezone issue! I somehow managed to forget I had set it to UTC some time ago via the following config:
```yaml
clickhouse:
      timezone: UTC
```
Hopefully we'll be able to update the default with that if it fixes the problem.

As for the other tables it seems Snuba creates a few of them for the various datasets, so it's probably worth creating distributed tables for each of them (I'll update the PR to do this)

**Comment by Mokto** [comment]
Perfect, thanks! After switching to UTC, I don't have any issue anymore <3 
I'm going to create a PR for Clickhouse, I'll let you handle Sentry jobs.

Can I ask why you chose to change to 2.0.0?

Is it really backward incompatible?

**Comment by J0sh0nat0r** [comment]
@Mokto Awesome! My concern with the backwards incompatibility is that some people may be running a version of Snuba released prior to getsentry/snuba#860 being merged, although if you consider updating the default Snuba tag and config simultaneously to not be breaking then a minor bump should suffice :)

**Comment by Mokto** [comment]
I don't have a strong opinion on this.

But yes, maybe putting 2.0 version is a good reminder for everyone to know they have to update the docker tags IF they forced them.

Let's do that!

**Comment by J0sh0nat0r** [comment]
@Mokto I've resolved the merge conflicts and tested this both locally and in production, should be good to go on your approval :)

**Comment by Mokto** [comment]
Also tried it, works fine! Thanks for your work!

**Pull request merged**